modules/data.atmosphere/R/check_met_input.R

```diff
@@ -0,0 +1,97 @@
+#' Check a meteorology data file for compliance with the PEcAn standard
+#'
+#' @param metfile Path of met file to check, as a scalar character.
+#' @param variable_table `data.frame` linking standard names to their
+#'   units. Defaults to [pecan_standard_met_table].
+#' @return `TRUE` (invisibly) if successful, otherwise throw an informative error.
+#' @author Alexey Shiklomanov
+#' @export
+check_met_input_file <- function(metfile, variable_table = pecan_standard_met_table) {
+
+  metfile <- normalizePath(metfile)
+
+  PEcAn.logger::severeifnot(
+    file.exists(metfile),
+    msg = glue::glue("File '{metfile}' does not exist.")
+  )
+
+  nc <- ncdf4::nc_open(metfile)
+
+  test_dims <- tryCatch(
+    testthat::test_that(
+      "Dimensions are correct.",
+      {
+        dimensions <- nc[["dim"]]
+        testthat::expect_type(dimensions, "list")
+        testthat::expect_equal(length(dimensions), 3)
+        testthat::expect_true("time" %in% names(dimensions))
```

Comment:
The problem I have with `$` subsetting is that it silently and "helpfully" accommodates incomplete names. So, for instance, `nc$nd` will happily return `3` because it completes to `ndim`. I think this behavior is maybe OK for interactive use, but I'd rather avoid it in rigorous tests or production code. In the past, I found a few places in PEcAn where `$` used the wrong name of a column or list element that didn't trigger an error because the new name was a superset of the old name (e.g. `days` vs. `days_in_year`).

Reply:
silent completion is a valid concern, though I submit it's less of a concern here because these names are controlled by the ncdf4 class, not assigned by a file's author. In fact I think I've answered my "is it safe..." question: Whichever one we actually check, I'd say one is enough and that if `length(dimensions) != nc$ndim` that means there's something wrong beyond the scope of our validation.